src/plugins/index_pattern_management/public/components/create_index_pattern_wizard/components/step_data_source/components/header/header.tsx

```diff
@@ -68,6 +70,12 @@ export const Header: React.FC<HeaderProps> = (props: HeaderProps) => {
       .then((fetchedDataSources: DataSourceTableItem[]) => {
         setIsLoading(false);
         if (fetchedDataSources?.length) {
+          fetchedDataSources = fetchedDataSources.filter((dataSource) =>
+            semver.satisfies(
+              dataSource.datasourceversion,
```

Comment:
what about for existing/old data sources that doesn't have this field? Does it break backwards compatibility?

Reply:
for older data sources it won't work yest as they won't support version decoupling, MDS initially just list out all data sources created regardless so there would be a bunch of unexpected behaviors, version decoupling is aiming to improve that user experiences by only allowing consumption against compatible data sources.

besides there is a `refresh` flow of data sources planning in progress which would also resolve the concern here, which will refresh the current saved object to bring it up-to-date